### ISSUE TITLE
Standardize Chat Prompt Templates to Use Jinja Format

### ISSUE THREAD

**Issue by calycekr** [huggingface/chat-ui#1525]
## Describe your feature request

Currently, the `chatPromptTemplate` for each model that can be set in env uses **Handlebars** format. However, the `chat_prompt` in the actual model's `tokenizer_config.json` uses **Jinja** format. This inconsistency is causing significant inconvenience. Since **Jinja** is widely used and preferred, it would be beneficial to standardize on **Jinja** format for both `chatPromptTemplate` and `chat_prompt`. This will improve consistency and ease of use for developers.

## Implementation idea

To implement this change, the following steps can be taken:

1. Update Codebase: Update the codebase to handle **Jinja** templates for `chatPromptTemplate`.

2. Documentation: Update the documentation to reflect this change and provide examples of how to use **Jinja** templates.

3. Testing: Thoroughly test the changes to ensure compatibility and that all existing templates work correctly with the new format.

**Comment by nsarrazin** [comment]
This is a good idea! It would be a breaking change though so we could keep handlebars as a fallback if the jinja evaluator fails, wdyt?

**Comment by julien-c** [comment]
yep, seems like the field has standardized on Jinja in the past year 😁 cc @Rocketknight1 😮

**Comment by julien-c** [comment]
cements that we're a Python-first company (and community) i guess 😁

**Comment by varad0309** [comment]
hey guys, did you add this feature?

**Comment by nsarrazin** [comment]
@varad0309 the recommended way for now would be to set `"tokenizer":` to point to the model's repo on the hub so it can pick up the chat template (using Jinja) there.

**Comment by varad0309** [comment]
Thanks @nsarrazin , thanks for the fast reply. Is there a way to do the same if we have a local model?
